utils/unminify.js

```diff
@@ -0,0 +1,145 @@
+ * references unminified.
+ */
+function unminifyLine(stackTraceLine, sourceMapConsumer) {
+  const urlRegex = /(https:(.*).js)/g;
+  const lineColumnNumberRegex = /:(\d+):(\d+)/g;
+  const lineColumnNumbers = lineColumnNumberRegex.exec(stackTraceLine);
+  const lineNumber = lineColumnNumbers[1];
+  const columnNumber = lineColumnNumbers[2];
+  const originalPosition = sourceMapConsumer.originalPositionFor({
+    line: parseInt(lineNumber),
+    column: parseInt(columnNumber),
+  });
+  stackTraceLine = stackTraceLine.replace(urlRegex, originalPosition.source);
+  const originalLocation = ':' + originalPosition.line + ':'
+      + originalPosition.column;
+  return stackTraceLine.replace(lineColumnNumbers[0], originalLocation);
+}
+
+/**
+ * @param {string} url
+ * @return {Promise} Promise that resolves to a source map.
+ */
+function getFromInMemory(url) {
+  return Promise.resolve(sourceMapConsumerCache.get(url));
+}
+
+/**
+ * @param {string} url
+ * @return {Promise} Promise that resolves to a source map.
+ */
+function getSourceMapFromNetwork(url) {
+  const reqPromise = new Promise((res, rej) => {
+    /**
+     * @param {error} err
+     * @param {response} _
+     * @param {body} body
+     */
+    function callback(err, _, body) {
+      if (err) {
+        rej(err);
+      } else {
+        try {
+          let sourceMapConsumer = new sourceMap.SourceMapConsumer(
+              JSON.parse(body));
+          requestCache.delete(url);
+          sourceMapConsumerCache.set(url, sourceMapConsumer);
+          res(sourceMapConsumer);
+        } catch (e) {
+          rej(e);
+        }
+      }
+    }
+    Request.request(url, callback);
+  });
+  requestCache.set(url, reqPromise);
+  return reqPromise;
+}
+
+/**
+ * @param {Array} sourceMapUrls
+ * @return {Array} Array of promises that resolve to source maps
+ */
+function extractSourceMaps(sourceMapUrls) {
+  const promises = [];
+  sourceMapUrls.forEach(function(sourceMapUrl) {
+    if (sourceMapConsumerCache.has(sourceMapUrl)) {
+      promises.push(getFromInMemory(sourceMapUrl));
+    } else if (requestCache.has(sourceMapUrl)) {
+      promises.push(requestCache.get(sourceMapUrl));
+    } else {
+      promises.push(getSourceMapFromNetwork(sourceMapUrl));
+    }
+  });
+  return promises;
+}
+
+/**
+ * @param {string} stackTrace
+ * @return {Promise} Promise that resolves to unminified stack trace.
+ */
+function unminify(stackTrace) {
+  const urlRegex = /(https:(.*).js)/g;
+  let match;
+  const stackTracesUrl = [];
+  while ((match = urlRegex.exec(stackTrace))) {
+    stackTracesUrl.push(match[0] + '.map');
+  }
+  const stackTraceLines = stackTrace.split('\n');
+  const promises = extractSourceMaps(stackTracesUrl);
+  return Promise.all(promises).then(function(values) {
+    let i = 0;
+    values.forEach(function(sourceMapConsumer) {
+      stackTraceLines[i] = unminifyLine(stackTraceLines[i],
+          sourceMapConsumer);
+      i++;
```

Comment:
`forEach()` gives you the index in the second param: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach